src/pages/PrepareCondition/index.tsx

```diff
@@ -162,21 +162,38 @@ export const PrepareCondition = () => {
   const [conditionType, setConditionType] = useState<ConditionType>(ConditionType.custom)
   const [outcomes, setOutcomes] = useState<Array<string>>([])
   const [outcome, setOutcome] = useState<string>('')
+  const [outcomesBeingEdited, setOutcomesBeingEdited] = useState<boolean[]>([])
 
   const addOutcome = useCallback(() => {
     const sanitizedOutcome = outcome.trim()
     const outcomesCloned = lodashClonedeep(outcomes)
     setOutcome('')
     setOutcomes([...outcomesCloned, sanitizedOutcome])
-  }, [outcome, outcomes, setOutcomes])
+    setOutcomesBeingEdited([...outcomesBeingEdited, false])
+  }, [outcome, outcomes, outcomesBeingEdited])
 
   const removeOutcome = useCallback(
     (index: number) => {
       const outcomesCloned = lodashClonedeep(outcomes)
       outcomesCloned.splice(index, 1)
       setOutcomes([...outcomesCloned])
+      setOutcomesBeingEdited([
+        ...outcomesBeingEdited.slice(0, index),
+        ...outcomesBeingEdited.slice(index + 1),
+      ])
     },
-    [outcomes]
+    [outcomes, outcomesBeingEdited]
+  )
+
+  const toggleEditOutcome = useCallback(
+    (value: boolean, index: number) => {
+      setOutcomesBeingEdited([
+        ...outcomesBeingEdited.slice(0, index),
+        value,
+        ...outcomesBeingEdited.slice(index + 1),
+      ])
+    },
+    [outcomesBeingEdited]
```

Comment:
Maybe, a better way to manage this is using an array of `Outcome = { value: string, editing: boolean}`